main.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"cloud.google.com/go/pubsub"
+)
+
+func main() {
+
+	err := publishMessage("pubsub1", "testing1", data{
+		messageID:    "1",
+		requestBody:  "api request body",
+		responseBody: "api response body",
+	})
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+}
+
+type data struct {
+	messageID		string
+	requestBody		string
+	responseBody	string
+}
+
+// PublishMessage publishes messages to already created topics
+func publishMessage(projectID, topicID string, msg data) error {
```

Comment:
Great work here. 

This function is an example of logic that does multiple things at once and could/should be broken down. 
I recommend 3 functions. 
- A function that creates a new pubsub client and then creates/returns the topic object.
- A function that accepts the topic object, and then uses it to construct and actually publish the message.

Reply:
This function should accept a context.Context. And the topic object, instead of the topic id. Let's have a separate function that actually generates the topic object

Reply:
Also, in go, the context.Context parameter should always be the first argument